---
title: "BBBQ 1"
author: Richel J.C. Bilderbeek
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, message=FALSE}
library(bbbq)
library(testthat)
library(tidyr)
library(ggplot2)
```

MHC-I haplotypes:

```{r}
print(get_mhc1_haplotypes())
```

Human samples:

```{r}
# n: sample size, equals the number of HLA haplotypes
n <- length(get_mhc1_haplotypes())
# f denotes 'fraction', a value between 0.0 and 1.0
f_human <- calc_overlaps(
  proteome_filename = "human.fasta",
  haplotypes = get_mhc1_haplotypes()
)
f_covid <- calc_overlaps(
  proteome_filename = "covid.fasta",
  haplotypes = get_mhc1_haplotypes()
)
f_myco <- calc_overlaps(
  proteome_filename = "myco.fasta",
  haplotypes = get_mhc1_haplotypes()
)
expect_equal(length(f_human), length(get_mhc1_haplotypes()))
expect_true(all(f_human > 0.0))
expect_true(all(f_human < 1.0))
expect_true(all(f_covid > 0.0))
expect_true(all(f_covid < 1.0))
expect_true(all(f_myco > 0.0))
expect_true(all(f_myco < 1.0))
```

Test human against COVID-19:

```{r}
t_human_covid <- ks.test(x = f_human, y = f_covid)
print(t_human_covid)
```

Test human against Mycobacterium:

```{r}
t_human_myco <- ks.test(x = f_human, y = f_myco)
print(t_human_myco)
```

Stats:

```{r}
alpha <- 0.05
get_verdict <- function(p_value, alpha) {
  if (p_value <= alpha) {
    return("Reject that the two samples are taken from a same distribution")
  }
  "Accept that the two samples are taken from a same distribution"
}
verdict_human_covid <- get_verdict(t_human_covid$p.value, alpha)
verdict_human_myco <- get_verdict(t_human_myco$p.value, alpha)
print(paste("COVID-19:", verdict_human_covid))
print(paste("Mycobacterium:", verdict_human_myco))
```

Create tidy table:

```{r}
df <- tibble::tibble(
  proteome = rep(c("human", "covid", "myco"), each = n),
  haplotype = rep(get_mhc1_haplotypes(), time = 3),
  f = c(f_human, f_covid, f_myco)
)
df$proteome <- as.factor(df$proteome)
df$haplotype <- as.factor(df$haplotype)
```

## Stats output

```{r}
df_stats_wide <- data.frame(
  statistic   = c(t_human_covid$statistic  , t_human_myco$statistic),
  p_value     = c(t_human_covid$p.value    , t_human_myco$p.value),
  alternative = c(t_human_covid$alternative, t_human_myco$alternative),
  method      = c(t_human_covid$method     , t_human_myco$method),
  data_name   = c(t_human_covid$data.name  , t_human_myco$data.name),
  alpha       = alpha,
  n           = length(get_mhc1_haplotypes()),
  verdict     = c(verdict_human_covid      , verdict_human_myco)
)
df_stats <- t(df_stats_wide)
values_covid <- df_stats[,1]
values_myco <- df_stats[,2]
param_names <- rownames(df_stats)
df_stats <- data.frame(
  parameter = param_names,
  value_covid = values_covid,
  value_myco = values_myco
)

knitr::kable(df_stats[,c(1,2)], row.names = FALSE)
knitr::kable(df_stats[,c(1,3)], row.names = FALSE)
```

```{r}
write.csv(df_stats[,c(1,2)], "bbbq_1_stats_covid.csv", row.names = FALSE)
write.csv(df_stats[,c(1,3)], "bbbq_1_stats_myco.csv", row.names = FALSE)
```

## Plot

```{r}
ggplot(df, aes(x = haplotype, y = f, fill = proteome)) + 
  scale_fill_manual(values = c("human" = "#ffffff", "covid" = "#cccccc", "myco" = "#999999")) +
  geom_col(position = position_dodge(), color = "#000000") + xlab("HLA haplotype") + 
  ylab("Epitopes overlapping \nwith transmembrane helix") +
  scale_y_continuous(labels = scales::percent_format(accuracy = 3)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_hline(yintercept = mean(f_human)) +
  geom_hline(yintercept = mean(f_covid), lty = "dashed") +
  geom_hline(yintercept = mean(f_myco), lty = "dotted") +
  labs(
    title = "MHC-I and its presentation of TMHs [THIS IS SIMULATED DATA]",
    caption = paste0(
      "Straight line: average in humans: ", formatC(100.0 * mean(f_human), digits = 3),"%. \n",
      "Dashed line: average in COVID-19: ", formatC(100.0 * mean(f_covid), digits = 3),"%. \n",
      "Dotted line: average in Mycoplasma: ", formatC(100.0 * mean(f_myco), digits = 3),"%."  
    )
  ) + ggsave("fig_bbbq_1.png", width = 7, height = 7)
```

## Table

Create wide (untidy) table:

```{r}
df_wide <- spread(df, proteome, f)
df_wide$covid <- formatC(100.0 * df_wide$covid, digits = 3)
df_wide$human <- formatC(100.0 * df_wide$human, digits = 3)
df_wide$myco <- formatC(100.0 * df_wide$myco, digits = 3)
```

Show table:

```{r}
knitr::kable(df_wide, caption = "Percentage of MHC-I epitopes overlapping with transmembrane helix")
```

Save table:

```{r}
write.csv(df_wide, "bbbq_1_percentages.csv", row.names = FALSE)
```


